compiler/optimizer/LocalOpts.cpp

```diff
@@ -6759,15 +6760,23 @@ static int32_t numSignatureChars(char *sig)
    return strchr(end,';')-sig+1;
    }
 
+void TR_InvariantArgumentPreexistence::traceIfEnabled(const char* format, ...)
```

Comment:
I see what you are trying to do here by simplifying the code, but there is a serious downside that I think needs to be considered. Under the old format of:
```
if (trace())
   traceMsg(....)
```
The evaluation of the arguments to the trace function was only done when tracing is enabled. With this change the arguments will always be evaluated (because C++ has eager argument evaluation) and so we will incur the cost of gathering all the strings etc needed for tracing even when the trace is not enabled.

I think the code should remain with the guards for the time being unless you can think of another implementation of this where we would not evaluate the arguments when doing the call...

Reply:
If you are going to refactor the tracing could you make it a separate commit from the functional change to make it easier to see what is going on in the history?

Reply:
Changed it back.